Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptDeleteController.java

```diff
@@ -129,17 +132,17 @@ public ModelAndView confirmlDelete(@PathVariable("id") String id,
         }
 
         conceptManager.deleteConcept(id, principal.getName());
-        List<ConceptEntry> founds = conceptManager.getConceptListEntries(concept.getConceptList());
-
-        foundConcepts = wrapperCreator
-                .createWrappers(founds != null ? founds.toArray(new ConceptEntry[founds.size()]) : new ConceptEntry[0]);
-
-        model.addObject("result", foundConcepts);
         if (fromHomeScreenDelete.equalsIgnoreCase("true")) {
+            List<ConceptEntry> founds = conceptManager.getConceptListEntries(concept.getConceptList(), 1, -1, "id",
+                    IConceptDBManager.DESCENDING);
+
+            foundConcepts = wrapperCreator.createWrappers(
+                    founds != null ? founds.toArray(new ConceptEntry[founds.size()]) : new ConceptEntry[0]);
+            model.addObject("result", foundConcepts);
             model.setViewName("redirect:/login");
```

Comment:
if you're redirecting here, I don't think the results will be forwarded to the redirect, will they? You probably have to add them as redirect attributes.